Code/GraphMol/SmilesParse/CXSmilesOps.cpp

```diff
@@ -420,6 +424,7 @@ bool parse_coords(Iterator &first, Iterator last, RDKit::RWMol &mol,
         }
         if (tokens.size() >= 3 && tokens[2].size()) {
           pt.z = boost::lexical_cast<double>(tokens[2]);
+          is3D = true;
```

Comment:
What if all pt.z are zero?

Reply:
Marvin doesn't write that, but it's worth checking like we do with the Mol file parser.
I'll update that.